src/FtpExtractor.php

```diff
@@ -62,26 +62,18 @@ public function copyFiles(string $sourcePath, string $destinationPath, FileState
             $this->logger->info('Connecting to host ...');
 
             (new RetryProxy(
-                new SimpleRetryPolicy(self::CONNECTION_RETRIES, [
-                    \LogicException::class,
-                    \RuntimeException::class,
-                    \ErrorException::class,
-                ]),
+                new SimpleRetryPolicy(self::CONNECTION_RETRIES),
                 new ExponentialBackOffPolicy(self::RETRY_BACKOFF),
                 $this->logger
             ))->call(static function () use ($adapter): void {
                 $adapter->getConnection();
             });
 
             $this->logger->info('Connection successful');
-        } catch (\RuntimeException $e) {
-            throw new UserException($e->getMessage(), $e->getCode(), $e);
-        } catch (\LogicException $e) {
-            throw new UserException($e->getMessage(), $e->getCode(), $e);
-        } catch (\ErrorException $e) {
-            throw new UserException($e->getMessage(), $e->getCode(), $e);
-        } catch (FileNotFoundException $e) {
-            throw new UserException($e->getMessage(), $e->getCode(), $e);
+        } catch (SftpAdapterException $e) {
```

Comment:
Tohle nějak nechápu - to by mělo být zabalené všechno vevnitř IMHO- prostě všechno padá do toho converteru a ten ví co s tím...

Reply:
Chyba na mé straně, samozřejmě máš pravdu, jsem to tam trochu pomotal :/